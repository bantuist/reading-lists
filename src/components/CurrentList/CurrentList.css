.Current-list {
  display: flex;
  flex-direction: column;
}
.Book-list {
  max-width: 900px;
  margin: 50px auto 0 auto;
  padding: 0;
  display: flex;
  justify-content: center;
  list-style-type: none;
  flex-wrap: wrap;
}
.Book-list-item {
  position: relative;
  margin: 10px;
}
.Book-list-item:hover {
  background-color: #382110;
}
.Book-list-item:hover img {
  opacity: .25;
}
.Book-list-item:hover .Book-details {
  color: #fff;
  display: block;
}
.Book-list-item:hover .Book-remove {
  color: #fff;
  display: block;
}
.image-container {
  height: 300px;
  width: 195px;
  overflow: hidden;
  border-radius: 1%;
  box-shadow: 3px 3px 3px #777;
}
.Book-list-item img {
  height: 100%;
}
.Book-details {
  font-size: 16px;
  display: none;
  position: absolute;
  top: 0;
  padding: 20px;
  height: 200px;
}
.Book-title {
  font-size: 20px;
}

.Book-remove {
  /*padding: 10px;*/
  font-size: 16px;
  display: none;
  background: transparent;
  border: none;
  position: absolute;
  top: 0;
  padding: 10px;
  right: 0;
}
.fa-check-square, .fa-check-square-o {
  cursor: pointer;
  display: none;
  color: #fff;
  font-size: 40px;
  position: absolute;
  top: 80%;
  right: 40%;
}
.Book-list-item:hover .fa {
  display: block;
  /*color: #382110;*/
}
