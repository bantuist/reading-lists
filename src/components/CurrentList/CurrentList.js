import React, { Component } from 'react';
// import PropTypes from 'prop-types';
import base from '../../rebase';
import { addBook, removeBook, updateBook } from '../edit-books';
import CurrentListItem from '../CurrentListItem/CurrentListItem';
import Search from '../Search/Search';
import './CurrentList.css';


class CurrentList extends Component {
  constructor() {
    super();
    this.state = {
      currentList: {},
    }

    this.addBook = this.addBook.bind(this);
    this.removeBook = this.removeBook.bind(this);
    this.updateBook = this.updateBook.bind(this);
  }
  componentWillMount() {
    let id = '';
    if (this.props.location) {
      id = this.props.location.state.currentList.id;
    } 

    // Update local state props whenever base prop changes
    base.bindToState(`/${id}`, {
      context: this,
      state: 'currentList',
      asArray: false,
    });
  }
  addBook(newBook) {
    let { currentList } = this.state;
    addBook(newBook, currentList);
  }
  removeBook(bookId) {
    const { currentList } = this.state;
    removeBook(bookId, currentList);
  }
  updateBook(book) {
    const { currentList } = this.state;
    updateBook(book, currentList);
  }

  render() {
    let { currentList } = this.state;
    let books = [];
    
    if (currentList.books) {
      books = Object.keys(currentList.books).map(key => {
        const book = currentList.books[key];
        const { id } = book;
        
        return (
          <CurrentListItem
            key={id}
            book={book}
            onUpdateBook={this.updateBook}
            onRemoveBook={this.removeBook}
          />
        );
      });
    }

    return (
      <div className="Current-list">
        <Search 
          currentList={currentList} 
          onAddBook={this.addBook}
        />
        <ul className="Book-list">
          {books}
        </ul>
      </div>
    );
  }
}

// CurrentList.propTypes = {
//   id: PropTypes.string.isRequired
// }

export default CurrentList;
