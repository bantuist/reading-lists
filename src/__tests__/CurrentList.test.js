import React from 'react';
import ReactDOM from 'react-dom';
import CurrentList from '../components/CurrentList/CurrentList';
import renderer from 'react-test-renderer';

it('should render CurrentList', () => {
  const tree = renderer.create(<CurrentList />).toJSON();
  expect(tree).toMatchSnapshot();
});