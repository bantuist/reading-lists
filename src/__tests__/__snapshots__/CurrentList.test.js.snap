// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CurrentList 1`] = `
<div
  className="Current-list"
>
  <div
    className="Search"
  >
    <div
      className="react-autosuggest__container"
    >
      <input
        aria-activedescendant={null}
        aria-autocomplete="list"
        aria-expanded={false}
        aria-haspopup={false}
        aria-owns="react-autowhatever-1"
        autoComplete="off"
        className="react-autosuggest__input"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="Add a book to undefined"
        role="combobox"
        type="text"
        value=""
      />
      <div
        className="react-autosuggest__suggestions-container"
        id="react-autowhatever-1"
      />
    </div>
    <div
      className="status"
    >
      <span
        style={
          Object {
            "paddingRight": "10px",
          }
        }
      >
        Requests: 
        0
        /1000
      </span>
      <span
        style={
          Object {
            "paddingRight": "10px",
          }
        }
      >
        Books: 
        /
         
      </span>
      Search for a book
    </div>
  </div>
  <ul
    className="Book-list"
  />
</div>
`;
